torch/csrc/autograd/profiler_utils.cpp

```diff
@@ -0,0 +1,96 @@
+#include <torch/csrc/autograd/profiler_utils.h>
+
+namespace torch { namespace autograd { namespace profiler {
+
+const size_t CONV2D_STRIDE = 3;
+const size_t CONV2D_PADDING = 4;
+const size_t CONV2D_DILATION = 5;
+const size_t CONV2D_GROUPS = 6;
+
+void saveExtraArgs(std::unordered_map<std::string, c10::IValue> &extra_args, const at::RecordFunction& fn) {
+    // switch the type of fn, if convolution or gemm, fill-in extra_args_, then return
+    std::vector<c10::IValue> inputs = fn.inputs();
+
+    TORCH_CHECK(
+      inputs.size() != 0,
+      "Tried to save extra arguments for flops computation, but input shape is not available!");
+
+    if(fn.name() == at::StringView("aten::conv2d")) {
+      at::Tensor input = inputs[0].toTensor();
```

Comment:
please make sure to check for out of bound access error, and make sure to check that ivalue is of the correct type (e.g. isTensor), if not we could print out a warning and refuse to set extra_args

Reply:
same below

Reply:
if there's an error, let's print a warning and let's not leave extra_args half-initialized

Reply:
use the return value (changing to bool) to signal the failure